File: packages/sdk-typescript/src/resources/billing.ts
// BillingResource — typed methods for /v1/billing (V-082).
//
// `getState` returns the current subscription mirror.
// `createCheckoutSession` returns a Stripe Checkout URL the customer
// redirects to. `createPortalSession` returns a Stripe Customer Portal
// URL. (The one-time trial_pack flow was retired 2026-05-27.)

import type {
  CreateCheckoutSessionRequest,
  CreateCheckoutSessionResponse,
  CreatePortalSessionResponse,
  GetBillingStateResponse,
} from '@driftstack/api-types';
import type { HttpClient } from '../http.js';

export class BillingResource {
  constructor(private readonly http: HttpClient) {}

  getState(): Promise<GetBillingStateResponse> {
    return this.http.request<GetBillingStateResponse>({
      method: 'GET',
      path: '/v1/billing',
    });
  }

  createCheckoutSession(
    body: CreateCheckoutSessionRequest,
  ): Promise<CreateCheckoutSessionResponse> {
    return this.http.request<CreateCheckoutSessionResponse>({
      method: 'POST',
      path: '/v1/billing/checkout-session',
      body,
    });
  }

  createPortalSession(): Promise<CreatePortalSessionResponse> {
    return this.http.request<CreatePortalSessionResponse>({
      method: 'POST',
      path: '/v1/billing/portal-session',
    });
  }
}

File: packages/sdk-go/billing.go
package driftstack

import "context"

// BillingResource handles /v1/billing endpoints (V-082).
//
// GetState returns the current subscription mirror state.
// CreateCheckoutSession returns a Stripe Checkout URL the customer
// redirects to. CreatePortalSession returns a Stripe Customer Portal
// URL. (The one-time trial_pack flow was retired 2026-05-27.)
type BillingResource struct {
	client *Client
}

// GetState returns the current subscription state.
func (r *BillingResource) GetState(ctx context.Context) (*GetBillingStateResponse, error) {
	var out GetBillingStateResponse
	if err := r.client.do(ctx, requestOptions{
		method: "GET",
		path:   "/v1/billing",
		out:    &out,
	}); err != nil {
		return nil, err
	}
	return &out, nil
}

// CreateCheckoutSession returns a Stripe Checkout URL for a tier
// subscription. The customer must be redirected to the URL to
// complete payment.
func (r *BillingResource) CreateCheckoutSession(ctx context.Context, body *CreateCheckoutSessionRequest) (*CreateCheckoutSessionResponse, error) {
	var out CreateCheckoutSessionResponse
	if err := r.client.do(ctx, requestOptions{
		method: "POST",
		path:   "/v1/billing/checkout-session",
		body:   body,
		out:    &out,
	}); err != nil {
		return nil, err
	}
	return &out, nil
}

// CreatePortalSession returns a Stripe Customer Portal URL for the
// current account. The customer manages payment method, invoices,
// and cancellation through the returned URL.
func (r *BillingResource) CreatePortalSession(ctx context.Context) (*CreatePortalSessionResponse, error) {
	var out CreatePortalSessionResponse
	if err := r.client.do(ctx, requestOptions{
		method: "POST",
		path:   "/v1/billing/portal-session",
		out:    &out,
	}); err != nil {
		return nil, err
	}
	return &out, nil
}
